# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsIFrame.h
# Commit: 6d76b71d89f3
# Full Hash: 6d76b71d89f3d3218632b68cecf9c7295b4fb41b
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2024-04-04 03:44:04
# Regressor Bug: 1887954
# File Overlap Count: 2
# Description:
#   Bug 1887954 - Make `nsIFrame::GetLastLeaf()` never returns a frame in native anonymous subtrees under the given frame r=emilio
#   
#   It does not check whether the child frame is in a native anonymous subtree
#   only when every first child at digging the frame tree.  This patch rewrites
#   the complicated loop with 2 for-loops and make it always check whether the
# ==============================================================================

diff -r 3187c4f491a8 -r 6d76b71d89f3 layout/generic/nsIFrame.h
--- a/layout/generic/nsIFrame.h	Wed Apr 03 07:15:55 2024 +0000
+++ b/layout/generic/nsIFrame.h	Wed Apr 03 07:20:57 2024 +0000
@@ -3939,6 +3939,9 @@
  public:
   // given a frame five me the first/last leaf available
   // XXX Robert O'Callahan wants to move these elsewhere
+  // FIXME: Only GetLastLeaf() never returns a leaf frame in native anonymous
+  // subtrees under aFrame.  However, GetFirstLeaf() may return a leaf frame
+  // in a native anonymous subtree.
   static void GetLastLeaf(nsIFrame** aFrame);
   static void GetFirstLeaf(nsIFrame** aFrame);
 